NoSoft-SA/nspack
# AG Grid version 28.1.0

Summary of this change:

Upgrade AG Grid from version 25.3.0 to version 28.1.0.
This version will allow for future use of sparkline columns and the server side row model.


CHANGELOG entry:

### Changed
- AG Grid upgraded to version 28.1.0

How to test this code via the UI:

1. Load a basic grid - does everything look normal?
1. Load a grouped grid - does everything look normal?
1. Load a tree grid - does everything look normal?
1. Run a search - does everything look normal?
1. Run a dataminer report - does everything look normal?
1. Move columns
1. Sort (click on headers)
1. Filter
1. Quick search
1. Use the columns toolbar on the right (hide/show, drag, re-order, group, sum, pivot)
1. Group
1. Action menus
1. Load a multiselect in a dialog and test that select/deselect, submit work correctly.
1. Load a multiselect in a page and test that select/deselect, submit work correctly.
1. Load a lookup grid and test that it works correctly
1. Use the view selected row feature (sort/un-sort columns, check display of values, use prev/next)
1. Check fullscreen mode (Note that context menus and column menus currently do not work in this mode)
1. Test export to CSV button
1. Use the hover button to jump to columns
1. Use the column state buttons and test that state is preserved/discarded correctly
1. Test grid's export to excel and csv
1. Select values in columns and check the counts/sum/avg etc in the bottom status bar
1. Test the grid context menu's copy functions
1. Test inline-edit columns - text, select, textarea
1. Test grid with coloured row text
